test/userslist.test.js

```diff
@@ -0,0 +1,103 @@
+const request = require('supertest'),
+  app = require('../app.js'),
+  validationErrorStatus = 401;
+
+describe('GET /users', () => {
+  describe('Test invalid inputs', () => {
+    test('Test invalid offset', () =>
+      request(app)
+        .get('/users')
+        .query({
+          token: 'token',
+          offset: -5,
+          limit: 10
+        })
+        .send()
+        .then(response => {
+          expect(response.status).toBe(validationErrorStatus);
+        }));
+
+    test('Test invalid limit', () =>
+      request(app)
+        .get('/users')
+        .send({
+          token: 'token',
+          offset: 10,
+          limit: -5
+        })
+        .then(response => {
+          expect(response.status).toBe(validationErrorStatus);
+        }));
+
+    test('Test invalid token', () =>
+      request(app)
+        .get('/users')
+        .send({
+          token: 'token',
+          offset: 10,
+          limit: 8
+        })
+        .then(response => {
+          expect(response.status).toBe(validationErrorStatus);
+        }));
+  });
+
+  describe('Test missing parameters', () => {
+    test.each([
+      {
+        token: 'token',
+        offset: 10
+      },
+      {
+        token: 'token',
+        limit: 10
+      },
+      {
+        offset: 10,
+        limit: 10
+      },
+      {}
+    ])('Test missing parameter with body = %p', body =>
+      request(app)
+        .get('/users')
+        .send(body)
+        .then(response => {
+          expect(response.status).toBe(validationErrorStatus);
+        })
+    );
+  });
+
+  describe('Test token validation', () => {
+    test('Create user, and ask users list with correct token', () =>
+      request(app)
+        .post('/users')
+        .send({
+          first_name: 'fn',
+          last_name: 'ln',
+          email: 'prueba@wolox.com.ar',
+          password: 'password'
+        })
+        .then(() =>
+          request(app)
+            .post('/users/sessions')
+            .send({
+              email: 'prueba@wolox.com.ar',
+              password: 'password'
+            })
+        )
+        .then(response =>
+          request(app)
+            .get('/users')
+            .query({
+              token: response.text,
+              offset: 0,
+              limit: 10
+            })
+            .send()
+        )
+        .then(response => {
+          expect(JSON.parse(response.text).rows.length).toBeGreaterThan(0);
```

Comment:
It's better to do `response.body.users` and then make a few more asserts with that. (Like checking the specific length it should have, or checking if the data on one of the users is correct)
All this assert tells me is that the endpoint responds with something, and nothing more. I like to think tests as a way of documenting the functionality, so they should cover more ground than this.